includes/Modules/Analytics_4.php

```diff
@@ -347,40 +355,49 @@ public function get_debug_fields() {
 				'debug' => Debug_Data::redact_debug_value( $settings['propertyID'], 7 ),
 			),
-			'analytics_4_web_data_stream_id' => array(
+			'analytics_4_web_data_stream_id'          => array(
 				'label' => __( 'Analytics 4 web data stream ID', 'google-site-kit' ),
 				'value' => $settings['webDataStreamID'],
 				'debug' => Debug_Data::redact_debug_value( $settings['webDataStreamID'] ),
 			),
-			'analytics_4_measurement_id'     => array(
+			'analytics_4_measurement_id'              => array(
 				'label' => __( 'Analytics 4 measurement ID', 'google-site-kit' ),
 				'value' => $settings['measurementID'],
 				'debug' => Debug_Data::redact_debug_value( $settings['measurementID'] ),
 			),
-			'analytics_4_use_snippet'        => array(
+			'analytics_4_use_snippet'                 => array(
 				'label' => __( 'Analytics 4 snippet placed', 'google-site-kit' ),
 				'value' => $settings['useSnippet'] ? __( 'Yes', 'google-site-kit' ) : __( 'No', 'google-site-kit' ),
 				'debug' => $settings['useSnippet'] ? 'yes' : 'no',
 			),
-		);
-
-		$debug_fields['analytics_4_available_custom_dimensions'] = array(
-			'label' => __( 'Analytics 4 available custom dimensions', 'google-site-kit' ),
-			'value' => empty( $settings['availableCustomDimensions'] )
-				? __( 'None', 'google-site-kit' )
-				: join(
-					/* translators: used between list items, there is a space after the comma */
-					__( ', ', 'google-site-kit' ),
-					$settings['availableCustomDimensions']
-				),
-			'debug' => empty( $settings['availableCustomDimensions'] )
-				? 'none'
-				: join( ', ', $settings['availableCustomDimensions'] ),
+			'analytics_4_available_custom_dimensions' => array(
+				'label' => __( 'Analytics 4 available custom dimensions', 'google-site-kit' ),
+				'value' => empty( $settings['availableCustomDimensions'] )
+					? __( 'None', 'google-site-kit' )
+					: join(
+						/* translators: used between list items, there is a space after the comma */
+						__( ', ', 'google-site-kit' ),
+						$settings['availableCustomDimensions']
+					),
+				'debug' => empty( $settings['availableCustomDimensions'] )
+					? 'none'
+					: join( ', ', $settings['availableCustomDimensions'] ),
+			),
+			'adsense_linked'                          => array(
+				'label' => __( 'AdSense Linked', 'google-site-kit' ),
+				'value' => $adsense_connected && $settings['adSenseLinked'] ? __( 'Connected', 'google-site-kit' ) : __( 'Not connected', 'google-site-kit' ),
+				'debug' => Debug_Data::redact_debug_value( $settings['adSenseLinked'] ),
+			),
+			'adsense_linked_last_synced_at'           => array(
+				'label' => __( 'AdSense Linked Last Synced At', 'google-site-kit' ),
+				'value' => $adsense_connected && $settings['adSenseLinkedLastSyncedAt'] ? gmdate( 'Y-m-d H:i:s', $settings['adSenseLinkedLastSyncedAt'] ) : __( 'Never synced', 'google-site-kit' ),
+				'debug' => Debug_Data::redact_debug_value( $settings['adSenseLinkedLastSyncedAt'] ),
+			),
 		);
```

Comment:
Let's add these two entries to the array only if both Analytics & AdSense are connected.

```suggestion
		);
		
		if ( $this->is_adsense_connected() ) {
			$debug_fields['adsense_linked'] = array(
				'label' => __( 'AdSense Linked', 'google-site-kit' ),
				'value' => $settings['adSenseLinked'] ? __( 'Connected', 'google-site-kit' ) : __( 'Not connected', 'google-site-kit' ),
				'debug' => Debug_Data::redact_debug_value( $settings['adSenseLinked'] ),
			);

			$debug_fields['adsense_linked_last_synced_at'] = array(
				'label' => __( 'AdSense Linked Last Synced At', 'google-site-kit' ),
				'value' => $settings['adSenseLinkedLastSyncedAt'] ? gmdate( 'Y-m-d H:i:s', $settings['adSenseLinkedLastSyncedAt'] ) : __( 'Never synced', 'google-site-kit' ),
				'debug' => Debug_Data::redact_debug_value( $settings['adSenseLinkedLastSyncedAt'] ),
			);
		}
```